src/pathfinding_igvc/test/path-finder-test.cpp

```diff
@@ -8,6 +8,148 @@
 #include <PathFinder.h>
 #include <gtest/gtest.h>
 
+TEST(PathFinder, TestPathWithNoObstacle) {
+    /* origin of OccupancyGrid */
+    // initialize origin of occupancy grid
+    geometry_msgs::Pose origin =
+            PathFinderTestUtils::constructPose(3.0, 3.0, 0.0);
+
+    /* mapMetaData of OccupancyGrid */
+    // initialize mapMetaData
+    nav_msgs::MapMetaData mapMetaData;
+    mapMetaData.resolution = 2.0;
+    mapMetaData.width = 2;
+    mapMetaData.height = 2;
+    // add origin to mapMetaData
+    mapMetaData.origin = origin;
+
+    /* OccupancyGrid */
+    // initialize occupancy grid
+    nav_msgs::OccupancyGrid grid;
+
+    // set mapMetaData
+    grid.info = mapMetaData;
+    grid.data = std::vector<int8_t>(4, AStar::GRID_FREE);
+
+    geometry_msgs::Point start;
+    start.x = 3.0;
+    start.y = 3.0;
+    geometry_msgs::Point goal;
+    goal.x = 5.0;
+    goal.y = 5.0;
+
+    nav_msgs::Path path = PathFinder::calculatePath(start, goal, grid);
+
+    ASSERT_EQ(path.poses.size(), 2);
+
+    EXPECT_FLOAT_EQ(path.poses[0].pose.position.x, 3.0);
+    EXPECT_FLOAT_EQ(path.poses[0].pose.position.y, 3.0);
+
+    EXPECT_FLOAT_EQ(path.poses[1].pose.position.x, 5.0);
+    EXPECT_FLOAT_EQ(path.poses[1].pose.position.y, 5.0);
+}
+
+TEST(PathFinder, TestFullPath) {
+    /* origin of OccupancyGrid */
+    // initialize origin of occupancy grid
+    geometry_msgs::Pose origin =
+            PathFinderTestUtils::constructPose(0.0, 0.0, 0.0);
+
+    /* mapMetaData of OccupancyGrid */
+    // initialize mapMetaData
+    nav_msgs::MapMetaData mapMetaData;
+    mapMetaData.resolution = 1.0;
+    mapMetaData.width = 10;
+    mapMetaData.height = 9;
+    // add origin to mapMetaData
+    mapMetaData.origin = origin;
+
+    /* OccupancyGrid */
+    // initialize occupancy grid
+    nav_msgs::OccupancyGrid grid;
+
+    // set mapMetaData
+    grid.info = mapMetaData;
+    grid.data = {
+            AStar::GRID_FREE, AStar::GRID_FREE, AStar::GRID_FREE, AStar::GRID_OCCUPIED, AStar::GRID_OCCUPIED, AStar::GRID_OCCUPIED, AStar::GRID_FREE, AStar::GRID_OCCUPIED, AStar::GRID_OCCUPIED, AStar::GRID_FREE,
+            AStar::GRID_FREE, AStar::GRID_OCCUPIED, AStar::GRID_FREE, AStar::GRID_FREE, AStar::GRID_FREE, AStar::GRID_FREE, AStar::GRID_OCCUPIED, AStar::GRID_FREE, AStar::GRID_FREE, AStar::GRID_FREE,
```

Comment:
Similar comment to above about a potential way to make these sorts of tests easier to read.